packages/apputils-extension/src/index.ts

```diff
@@ -90,10 +91,60 @@ const paletteRestorer: JupyterFrontEndPlugin<void> = {
 const settings: JupyterFrontEndPlugin<ISettingRegistry> = {
   id: '@jupyterlab/apputils-extension:settings',
   activate: async (app: JupyterFrontEnd): Promise<ISettingRegistry> => {
-    const connector = app.serviceManager.settings;
-    const plugins = (await connector.list()).values;
+    const { isDeferred, isDisabled } = PageConfig.Extension;
+    const connector = new (class SettingConnector extends DataConnector<
+      ISettingRegistry.IPlugin,
+      string
+    > {
+      fetch(id: string): Promise<ISettingRegistry.IPlugin> {
+        return app.serviceManager.settings.fetch(id);
+      }
+
+      async list(
+        query: 'active' | 'all' = 'all'
+      ): Promise<{ ids: string[]; values: ISettingRegistry.IPlugin[] }> {
+        let { ids, values } = await app.serviceManager.settings.list();
+
+        if (query === 'all') {
+          return { ids, values };
+        }
+
+        return {
+          ids: ids.filter(id => !isDeferred(id) && !isDisabled(id)),
+          values: values.filter(({ id }) => !isDeferred(id) && !isDisabled(id))
+        };
+      }
+    })();
+
+    const registry = new SettingRegistry({
+      connector,
+      plugins: (await connector.list('active')).values
+    });
+
+    // If there are plugins that have schemas that are not in the setting
+    // registry after the application has restored, try to load them manually.
```

Comment:
Just so we're clear: what is the purpose of the code that follows? This comment says what is happening, but not why.

Reply:
Basically, if a plugin is deferred but not disabled and nobody has chosen to use it, its settings will not be editable by end users. This logic is meant to handle that scenario. It also handles the problem you originally pointed out if a deferred plugin has a transformation that never completes.